File: 48-rotate-image/48-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& m) {
        int rows = m.size(), cols = m[0].size();
        
        for(int i = 0; i < rows; ++i){
            for(int j = 0; j < cols; ++j){
                if(i == j) continue;
                else if(i < j)
                    swap(m[i][j], m[j][i]);
            }
        }
        
        for(int i = 0; i < rows; ++i){
            for(int j = 0; j < cols; ++j){
                if(j < cols / 2)
                    swap(m[i][j], m[i][cols - j - 1]);
            }
        }
    }
};

File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows = len(matrix)
        cols = len(matrix[0])
        
        for i in range(rows):
            for j in range(cols):
                if(i < j):
                    temp = matrix[i][j]
                    matrix[i][j] = matrix[j][i]
                    matrix[j][i] = temp
        
        for i in range(rows):
            for j in range(int(cols/2)):
                temp = matrix[i][j]
                matrix[i][j] = matrix[i][cols - j - 1]
                matrix[i][cols - j - 1] = temp